src/wwwroot/app/views/vacancy/vacancy.edit.controller.js

```diff
@@ -166,4 +170,17 @@ export default function VacancyController( //eslint-disable-line max-statements
    function _getStateTitle(key) {
       return $translate.instant(key);
    }
+
+// The first location of the vacancy should match the responsible user's location
+   function _vacacyEditValidation(vacancy) {
+      if (vacancy.cityIds[0] !== vacancy.responsible.cityId) {
```

Comment:
you check it before standart validation, so it is possible to be that cityIds will be empty or undefined. Please, check yhis case, and by results, may be, add condition for this case